rdflib/plugins/memory.py

```diff
@@ -49,6 +52,15 @@ def add(self, triple, context, quoted=False):
         # exits.
         subject, predicate, object = triple
         spo = self.__spo
+
+        try:
+            # Triple already in store
+            if(spo[subject][predicate][object]==1):
+                pass
```

Comment:
I think you could actually `return` here, rather than `pass`.
After spending the cpu time to check if the triple already exists in the index, we should at least take advantage of that answer and exit the method early if we know it exists.
(This assumes that if it exists in the `spo` index, then it also exists in the `pos` and the `osp` indexes)

Reply:
Your assumption makes sense, I did not think of this. I'll update the code. Thanks!